mm/2s2h/Enhancements/Equipment/InstantRecall.cpp

```diff
@@ -0,0 +1,44 @@
+#include <libultraship/bridge.h>
+#include "Enhancements/GameInteractor/GameInteractor.h"
+#include "InstantRecall.h"
+#include "src/overlays/actors/ovl_En_Boom/z_en_boom.h"
+
+extern "C" {
+extern PlayState* gPlayState;
+}
+
+static HOOK_ID onActorUpdateHookId = 0;
+
+void Player_ReturnBoomerangs() {
+    Player* player = GET_PLAYER(gPlayState);
+
+    EnBoom* boomerangs = (EnBoom*)player->boomerangActor;
+
+    if (boomerangs != NULL) {
+        // Set the positions of both boomerangs
+        boomerangs->actor.world.pos = player->actor.world.pos;
+        if (boomerangs->actor.child != NULL) {
+            boomerangs->actor.child->world.pos = player->actor.world.pos;
+        }
+
+        // Kill both boomerangs
+        Actor_Kill(&boomerangs->actor);
+        if (boomerangs->actor.child != NULL) {
+            Actor_Kill(boomerangs->actor.child);
+        }
+    }
+}
+
+void RegisterInstantRecall() {
+    GameInteractor::Instance->UnregisterGameHookForID<GameInteractor::OnActorUpdate>(onActorUpdateHookId);
+    onActorUpdateHookId = 0;
+
+    onActorUpdateHookId = GameInteractor::Instance->RegisterGameHookForID<GameInteractor::OnActorUpdate>(
+        ACTOR_EN_BOOM, [](Actor* outerActor) {
+            if (CVarGetInteger("gEnhancements.PlayerActions.InstantRecall", 0)) {
+                if (gPlayState->state.input->cur.button == BTN_B) {
```

Comment:
this should be using CHECK_BTN_ALL, otherwise this will only return true if only the b button and nothing else is being held. I'm wondering if it should be using press instead of cur too

```suggestion
                if (CHECK_BTN_ALL(gPlayState->state.input->cur.button, BTN_B)) {
```

Reply:
I agree, and yeah I think `press` would be better to use as well.

```suggestion
                if (CHECK_BTN_ALL(gPlayState->state.input->press.button, BTN_B)) {
```